app/javascript/components/users/UsersTable.vue

```diff
@@ -0,0 +1,139 @@
+      
+      <!-- Column template for Type -->
+      <template #cell(provider)="data">
+        {{ ldapColumn(data.item) }}
+      </template>
+
+      <!-- Column template for Role -->
+      <template #cell(role)="data">
+        <form :id="formId(data.item)" :action="formAction(data.item)" method="post">
+          <input type="hidden" name="_method" value="put" />
+          <input type="hidden" name="authenticity_token" :value="authenticityToken" />
+          <select class="form-control" name="user[admin]" @change="adminStatusChanged($event, data.item)" v-model="data.item.admin">
+            <option value="false">user</option>
+            <option value="true">admin</option>
+          </select>
+        </form>
+      </template>
+
+      <!-- Column template for Actions -->
+      <template #cell(actions)="data">
+        <b-button class="float-right"
+                  variant="danger"
+                  data-confirm="Are you sure you want to permanently remove this user?" 
+                  data-method="delete" 
+                  :href="formAction(data.item)"
+                  rel="nofollow">
+          <i class="mdi mdi-trash-can" aria-hidden="true"></i>
+          Remove
+        </b-button>
+      </template>
+    </b-table>
+
+    <!-- Pagination controls -->
+    <b-pagination
+      v-model="currentPage"
+      :total-rows="rows"
+      :per-page="perPage"
+      aria-controls="users-table"
+    ></b-pagination>
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'UsersTable',
+  props: {
+    users: {
+      type: Array,
+      required: true,
+    }
+  },
+  data: function () {
+    return {
+      search: "",
+      perPage: 10,
+      currentPage: 1,
+      fields: [
+        { key: 'name', label: 'User' },
+        { key: 'provider', label: 'Type' },
+        'role',
+        { key: 'actions', label: '' }
+      ]
+    }
+  },
+  computed: {
+    // Search users based on name and email
+    searchedUsers: function () {
+      let downcaseSearch = this.search.toLowerCase()
+      return this.users.filter(user => user.email.toLowerCase().includes(downcaseSearch) || user.name.toLowerCase().includes(downcaseSearch));
+    },
+    // Used by b-pagination to know how many total rows there are
+    rows: function() {
+      return this.searchedUsers.length;
+    },
+    // Authenticity Token for forms
+    authenticityToken: function() {
+      return document.querySelector("meta[name='csrf-token']").getAttribute("content");
+    },
+    // Total number of users in the system
+    userCount: function() {
+      return this.users.length;
+    }
+  },
+  methods: {
+    // Automatically submit the form when a user selects a form option
+    adminStatusChanged: function(event, user) {
+      document.getElementById(this.formId(user)).submit();
+    },
+    // The text that should appear in the 'Type' column
+    ldapColumn: function(user) {
+      return user.provider === null ? 'Vulcan User' : 'LDAP User'
```

Comment:
Update to say local user instead of vulcan user based on today's feedback mtg

Reply:
pushed a change for this